mambaTests/HLSMediaSpanTests.swift

```diff
@@ -114,4 +114,46 @@ class HLSMediaSpanTests: XCTestCase {
         let hlsString = hlsArray.joined()
         runTest(hlsString: hlsString, expectedSpans: [0...1, 2...4, 5...8])
     }
+    
+    // This validates the early return logic in generateMediaSpans() for empty mediaSegmentGroups.
+    func testNoMediaSegmentsScenario() {
+        let hlsArray = [
+            "#EXTM3U\n",
```

Comment:
I guess personal preference but I think triple-quoted strings are easier to read and don't need to be joined.
i.e. 
```swift
let testHls = """
#EXTM3U
#EXT-X-VERSION:3 // etc
"""
```